lib/parser.dart

```diff
@@ -259,9 +259,9 @@ class _Parser {
   }
 
   Token _next({bool unicodeRange = false}) {
-    _previousToken = _peekToken;
+    final next = _previousToken = _peekToken;
```

Comment:
It desugars to something with a `let` I think, so the closest in actual Dart is

```dart
final tmp = _peekToken;
_previousToken = tmp;
next = tmp;
```

In any case, my understanding is that the promotion behavior here _is_ expected to be something we can rely on, and I personally find this version more readable than the manual desugaring (even the one that skips the unnecessary extra var) despite being a bit more "tricky".

cc @leafpetersen in case I'm mistaken about the promotion here.

Reply:
Actually ya the way you describe it does make more sense - in the case of a getter you would want to ensure that `_peekToken` is only read a single time I think.

Reply:
More like the latter "desugaring", but either way the type of the initializer is the type of the assignment expression, which is the type of the rhs:

```
  // No errors
  Object? x;
  (x = 3).isEven;
```